homeassistant/data_entry_flow.py

```diff
@@ -201,11 +201,14 @@ def async_has_matching_flow(
         If match_context is passed, only return flows with a context that is a
         superset of match_context.
         """
-        return any(
-            flow
-            for flow in self._async_progress_by_handler(handler, match_context)
-            if flow.init_data == data
-        )
+        match_context_items = match_context.items()
+        for progress in self._handler_progress_index.get(handler, ()):
+            if (
+                match_context_items <= progress.context.items()
+                and progress.init_data == data
+            ):
+                return True
```

Comment:
Avoids looping the data twice since there are some of these that generate large enough data sets that the system gets a bit wobbly at the started event when the herd of discoveries happens